src/MyModal.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+
+
+
+class MyModal extends React.Component {
+
+
+  listenKeyboard(event) {
+    if (event.key === 'Escape' || event.keyCode === 27) {
+      this.props.onClose();
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.onClose) {
+      window.addEventListener('keydown', this.listenKeyboard.bind(this), true);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.onClose) {
+      window.removeEventListener('keydown', this.listenKeyboard.bind(this), true);
+    }
+  }
```

Comment:
This is now a method that is 'unsafe' and will be supported up to a point, but react is starting to filter out the usage of this [method and some others](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount).